globals/matchmaking_globals.go

```diff
@@ -9,4 +11,60 @@ type CommonMatchmakeSession struct {
 	ConnectionIDs []uint32 //players in the room, referenced by their connection IDs. This is used instead of the PID in order to ensure we're talking to the correct client (in case of e.g. multiple logins)
 }
 
-var Sessions = []CommonMatchmakeSession{}
\ No newline at end of file
+var Sessions = []CommonMatchmakeSession{}
+
+func DeleteIndex(s []uint32, index int) []uint32 {
+    return append(s[:index], s[index+1:]...)
+}
+
+func RemoveConnectionIDFromRoom(clientConnectionID uint32, gathering uint32){
+	for index, connectionID := range Sessions[gathering].ConnectionIDs {
+		if connectionID == clientConnectionID {
+			Sessions[gathering].ConnectionIDs = DeleteIndex(Sessions[gathering].ConnectionIDs, index)
+		}
+	}
+	if len(Sessions[gathering].ConnectionIDs) == 0{
+		Sessions[gathering] = CommonMatchmakeSession{} //TODO: is there a better way to do this? right now, the index is used as the gathering id, and I'm not sure if there's a good way to remove without reordering.
+	}
+}
+
+func FindClientSessions(clientConnectionID uint32) []uint32 {
+	//TODO: is there any real instance where a client could be in multiple MatchmakeSessions at once? afaik there isn't, but I handle it regardless just in case (if we're sure there isn't, we could remove the handling and speed the code up)
+	gatheringsFoundIn := []uint32{}
+	for gatheringIndex, _ := range Sessions {
```

Comment:
If you only need the index, a standard `for` loop is likely more performant since `for range` already reads the value at the index during iteration, which is thrown out here

Reply:
One last thing and I think this is ready: does this apply now that we're using a map for Sessions? since the keys aren't gonna be in order every time anymore

Reply:
No, when using a map type this is the right way to go. `for range` doesn't return an index, it returns a key (it just so happens that with a slice those are the same thing), so this is the right way to go to get these keys

Reply:
Though I would say that the name `gatheringIndex` is misleading since it's no longer an index